var/spack/repos/builtin/packages/hypre/package.py

```diff
@@ -78,6 +78,21 @@ def install(self, spec, prefix):
             os.environ['CXX'] = spec['mpi'].mpicxx
             os.environ['F77'] = spec['mpi'].mpif77
             configure_args.append('--with-MPI')
+            # If using mpi, explicitly pass in compiler flags
```

Comment:
why do you need to do this? All that is done automatically through compiler wrappers, also in case of MPI as corresponding variables should be set to make MPI wrappers use Spack compiler wrappers. 

As far as I understand, this change shall not be necessary.

Reply:
@scheibelp @alalazo @tgamblin  ping.

Reply:
here's the relevant code for `openmpi`: https://github.com/spack/spack/blob/develop/var/spack/repos/builtin/packages/openmpi/package.py#L239-L242

Reply:
Perhaps this is because the build system wants to know the compiler flags at configure time (one example is that `configure` may choose to add compiler flags that conflict with the ones that Spack intends to inject at wrapper invocation time). @becker33 has a PR at https://github.com/spack/spack/pull/6415 to provide a means to do this with less work; for now though I think this is the way to do it.

@junkudo am I right that the compiler options need to be known for you at `configure` time? Out of curiosity does this need to be merged quickly or would you be interested in waiting for #6415?

Reply:
Thanks for the quick turn around!  
I see now that I failed to absorb the bit around 'Wrapping wrappers' in the documentation.  I only saw that the build output was calling my `mpicxx` (not a spack mpicxx) and incorrectly assumed it was not calling the spack compiler wrappers.  I wasn't even aware that one could re-point the underlying compilers of mpi.   It's pretty clear to me that things are working as intended when I install with the `-d` flag.  

As for needing compiler options at `configure` time, I don't currently have a use case that requires this; I think for me, it's more along the lines of that I'm used to seeing the `configure` steps verifying and printing inputted flags.  Flags at the `configure` step would however avoid the issues of possible default optimization flags over-riding the spack injected ones.